lib/flatpak-installation.c

```diff
@@ -1361,7 +1361,12 @@ flatpak_installation_install_full (FlatpakInstallation    *self,
   else
     ostree_progress = ostree_async_progress_new_and_connect (no_progress_cb, NULL);
 
-  if (!flatpak_dir_install (dir_clone, FALSE, FALSE,
+  /* Note that if the caller sets FLATPAK_INSTALL_FLAGS_NO_DEPLOY this
+   * function will always return an error, because get_ref will return NULL
+   * later (as there is no installed ref yet). */
```

Comment:
We could at least add a custom FlatpakError version from this so the caller can detect this error.

Reply:
In this case we return `FLATPAK_ERROR_APP_NOT_INSTALLED`. It is *technically* unique, but I suppose to avoid ambiguity we could use something like `FLATPAK_ERROR_APP_ONLY_PULLED`. Thoughts?

Reply:
Yeah, i would prefer FLATPAK_ERROR_APP_ONLY_PULLED.

Reply:
Okay, added FLATPAK_ERROR_ONLY_PULLED (and an explanation in the function docs as to the expected behaviour in this case).